backend/protocol_rpc/message_handler/fastapi_handler.py

```
@@ -0,0 +1,196 @@
+        new_msg_handler.client_session_id = client_session_id
+        return new_msg_handler
+    
+    def log_endpoint_info(self, func):
+        """Decorator for logging endpoint information."""
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            # Log endpoint call
+            logger.info(f"Endpoint called: {func.__name__}")
+            try:
+                result = func(*args, **kwargs)
+                return result
```

Ruff (0.13.3)

Consider moving this statement to an `else` block

(TRY300)